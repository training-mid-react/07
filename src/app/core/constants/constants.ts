
export const ROWS = 6;
export const COLS = 7;
export const WINNING_LENGTH = 4;

export const PLAYER1 = {
    id: 1,
    name: "Player 1",
    color: "#fee813",
};
export const PLAYER2 = {
    id: 2,
    name: "Player 2",
    color: "#fe4c13",
};

