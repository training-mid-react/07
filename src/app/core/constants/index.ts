export const ROWS = 6;
export const COLS = 7;

export const BASE_MATRIZ = new Array(ROWS + 1)
    .fill(null)
    .map(() => new Array(COLS).fill(null));
