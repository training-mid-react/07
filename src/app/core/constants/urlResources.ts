import { environment } from "../../../environment/environment";

export const urlResources = {
  user: `${environment.apiUrl}/user`,
  getPokemon: (pokemon: string) => `${environment.apiUrl}/pokemon/${pokemon}`,
};