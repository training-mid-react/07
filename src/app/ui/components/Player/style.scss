.player {
    h2 {
        color: #fff;
        text-align: center;
        margin-left: 5px;
    }
    
    &__flex {
        display: flex;
        align-items: center;
        justify-content: center;
        margin-bottom: 20px;
    }
}