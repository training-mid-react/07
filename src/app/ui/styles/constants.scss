$primary-color: #14006d;
$secondary-color: #ff0066;
$background-color: #000000;
$text-color: #ffffff;
