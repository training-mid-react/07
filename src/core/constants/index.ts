export const directions = [
  { row: 0, col: 1 },
  { row: 1, col: 0 },
  { row: 1, col: 1 },
  { row: 1, col: -1 },
];
