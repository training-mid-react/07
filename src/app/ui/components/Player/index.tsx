import { IPlayer } from "@/app/core/interfaces/player.interface";
import { InputText } from "../../forms/inputText";
import "./style.scss"
import { InputColor } from "../../forms/inputColor";

interface Props {
    player: IPlayer;
    value: string;
    onChangeInput: (value: string) => void;
    handleBlurInput: () => void
}

const Player = ({ player, value, onChangeInput, handleBlurInput }: Props) => {
    return (
        <article className="player">
            <div className="player__flex">
                <InputColor color={player.color} isDisabled={true} />
                <h2>{player.name}</h2>
            </div>
            <InputText value={value} onChange={onChangeInput} placeholder="Escribe tu nombre" handleBlurInput={handleBlurInput} />
        </article>
    );
}

export default Player;