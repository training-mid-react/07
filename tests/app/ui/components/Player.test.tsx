import { fireEvent, render, screen } from "@testing-library/react";
import React from "react";
import { vi } from "vitest";
import { PLAYER1 } from "../../../../src/app/core/constants/constants";
import { IPlayer } from "../../../../src/app/core/interfaces/player.interface";
import Player from "../../../../src/app/ui/components/Player";

describe("Test de Player", () => {
    const mockHandleChangeInput = vi.fn();
    const mockHandleBlurInput = vi.fn();
    const player: IPlayer = PLAYER1;
    const newValue = "Cristian";

    test("Snapshot del Player", () => {
        const { asFragment } = render(<Player player={player} value={newValue} onChangeInput={mockHandleChangeInput} handleBlurInput={mockHandleBlurInput} />)
        expect(asFragment).toMatchSnapshot();
    })

    test("Renderiza el Nombre del Player correctamente", () => {
        render(<Player player={player} value={newValue} onChangeInput={mockHandleChangeInput} handleBlurInput={mockHandleBlurInput} />)
        const playerName = screen.getByText("Player 1");
        expect(playerName).toBeTruthy();
        
    })

    test("Renderiza el Player con el color correcto", () => {
        const { container } = render(<Player player={player} value={newValue} onChangeInput={mockHandleChangeInput} handleBlurInput={mockHandleBlurInput} />)        
        const input = container.querySelector(".player input");
        const color = input?.getAttribute("value");
        expect(color).toBe(player.color);
    })
    test("Llamar al metodo onChangeInput cuando el input cambia", () => {
        render(
            <Player
              player={player}
              value={newValue}
              onChangeInput={mockHandleChangeInput}
              handleBlurInput={mockHandleBlurInput}
            />
          );
      
          // Simula un cambio en el input
          const inputField = screen.getByPlaceholderText("Escribe tu nombre");
          
          fireEvent.change(inputField, { target: { value: "Nuevo nombre del jugador" } });
      
          // Verifica que onChangeInput haya sido llamado con el nuevo valor
          expect(mockHandleChangeInput).toHaveBeenCalledWith('Nuevo nombre del jugador');
          expect(mockHandleChangeInput).toHaveBeenCalledTimes(1);
    })
})