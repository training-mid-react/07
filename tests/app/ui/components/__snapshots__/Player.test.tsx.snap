// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test de Player > Snapshot del Player 1`] = `[Function]`;
